File: SchoolManagementApi/DTOs/Students/StudentResponseDto.cs
namespace SchoolManagementApi.DTOs.Students;

public class StudentResponseDto
{
    public int Id { get; set; }
    public string Name { get; set; } = null!;
    public string Surname { get; set; } = null!;
    public string Email { get; set; } = null!;

    public string FullName => $"{Name} {Surname}";
}

File: SchoolManagementApiNode/src/students/dto/student-response.dto.ts
import { ApiProperty } from '@nestjs/swagger';

export class StudentResponseDto {
  @ApiProperty()
  id: number;

  @ApiProperty()
  name: string;

  @ApiProperty()
  surname: string;

  @ApiProperty()
  email: string;

  @ApiProperty()
  fullName: string;
}
